posthog/api/capture.py

```diff
@@ -191,25 +83,35 @@ def get_event(request):
         return cors_response(request, JsonResponse({'code': 'validation', 'message': "No api_key set. You can find your API key in the /setup page in posthog"}, status=400))
 
     try:
-        team = Team.objects.get(api_token=token)
+        team = Team.objects.only('pk').get(api_token=token)
     except Team.DoesNotExist:
         return cors_response(request, JsonResponse({'code': 'validation', 'message': "API key is incorrect. You can find your API key in the /setup page in PostHog."}, status=400))
 
     if isinstance(data, dict):
         if data.get('batch'): # posthog-python and posthog-ruby
-            data = data['batch']
-            assert data is not None
+            events = data['batch']
+            assert events is not None
         elif 'engage' in request.path_info: # JS identify call
             data['event'] = '$identify' # make sure it has an event name
 
     if isinstance(data, list):
```

Comment:
Found breaking while testing QA'ing endpoint. Logic changed here because of variable renaming and we're not calling process_event directly inside if statements. I believe the above `events = data['batch']` and  `assert events is not None` should use the `data` variable again so that it's properly passed downwards into this if statement. Right now, the above could possibly set the `events` variable but then we check `data` again inside this if/statement which resets `events` before continuing below

Reply:
Absolutely right. Little over-zealous with removing tests. Fixed now!